### DISCUSSION TITLE
Safe Exam Browser 3.3.3 not available anymore

### DISCUSSION THREAD

**Discussion by Karelbe** [SafeExamBrowser/seb-mac#456] [Q&A] [answered]
I noticed today that the SEB 3.3.3 installer for macOS has been removed from the repository. Unfortunately, we can’t upgrade to 3.4 yet, as it hasn’t been fully tested with Ans (our online assessment tool). I’ve done some preliminary tests myself, and several issues arose, making 3.4 unusable for us at the moment.

Could you clarify why the 3.3.3 installer was removed?

**Comment by danschlet** [comment]
SEB 3.3.3 is leading to crashes in browsers like Chome and derived versions (Edge, Opera, Brave) and Firefox. As these seem to affect hundred thousands or millions of users, we feel like it's urgent to prevent SEB 3.3.3 to be further downloaded and used.

In general, our goal is to only make one latest SEB version available, as we constantly have to improve security and sometimes fix critical issues like this one. That seemed to work well with iOS (as there's always only the latest version available in the App Store), so we assumed this won't be critical on macOS as well. 

We usually provide beta and release candidate versions in the [Releases](https://github.com/SafeExamBrowser/seb-mac/releases) section, which you can add to your GitHub watchlist and get notified when such pre-release versions are available. **It's crucial to test pre-release versions asap** and report issued, **before** the final release is out. Most of our partners which are contributors to the [SEB Alliance](https://safeexambrowser.org/alliance/members.html) do this and are more involved in the pre-release testing process.

Which issues arose with SEB 3.4 which make it "unusable" for you? Did you already create issues here on GitHub for those?

**Reply by Karelbe** [reply]
I’ve only done some preliminary testing with the SEB integration in Ans. Since SEB is configured within Ans, I cannot enable extensive logging, so I’ve opened a case with them. Ans mentioned that they haven’t yet had time to test SEB 3.4.

The issue we’re encountering is that when starting a test (and thus SEB), an error appears in Ans: “Incorrect configuration detected - Exit the Safe Exam Browser.” In attachment you can find the config file that Ans generates (It's downloadable in the interface)
[ans_seb_config.zip](https://github.com/user-attachments/files/17648680/ans_seb_config.zip)

**Reply by danschlet** [reply, marked as answer]
The config file looks very basic and doesn't seem to contain anything I could imagine causing issues with SEB 3.4. I wonder if they are checking the BrowserExamKey, which also contains a hash representing the app code signature checksum (that one is different for any version/build of SEB). I guess only Ans will be able to find out what exactly the issue is.

I will discuss with Google if they can prevent those crashes in Chrome when SEB is run also from their side. Also I need to fix some unrelated issues in SEB 3.4 and release a 3.4.1 patch update. 

In the meantime I made SEB 3.3.3 available again on GitHub at this path: https://github.com/SafeExamBrowser/seb-mac/releases/download/3.3.3/SafeExamBrowser-3.3.3_deprecated.dmg

**Reply by Karelbe** [reply]
Thank you Daniel. I also assume the issue is with Ans and how they built the integration with SEB. I have urged them again to address this (since we are currently in mid-terms) .
